lib/core/include/resource_administration.hpp

```diff
@@ -0,0 +1,147 @@
+#undef RxComm
+
+#ifdef IRODS_RESOURCE_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+    #define NAMESPACE_IMPL      server
+    #define RxComm              RsComm
+
+    struct RsComm;
+#else
+    #define NAMESPACE_IMPL      client
+    #define RxComm              RcComm
+
+    struct RcComm;
+#endif // IRODS_RESOURCE_ADMINISTRATION_ENABLE_SERVER_SIDE_API
+
+#include <cstdint>
+#include <string>
+#include <string_view>
+#include <system_error>
+#include <tuple>
+#include <optional>
+#include <chrono>
+
+namespace irods::experimental::administration
+{
+    // clang-format off
+    using resource_name_type = std::string_view;
+    using resource_time_type = std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds>;
+    // clang-format on
+
+    // Forward declaration for "info" API function.
+    class resource_info;
+
+    // Forward declaration for "resource_info" class.
+    namespace NAMESPACE_IMPL
+    {
+        auto info(RxComm&, const resource_name_type) -> std::tuple<std::error_code, std::optional<resource_info>>;
+    } // namespace NAMESPACE_IMPL
+
+    namespace resource_type
+    {
+        // clang-format off
+        const std::string_view compound       = "compound";
+        const std::string_view deferred       = "deferred";
+        const std::string_view load_balanced  = "load_balanced";
+        const std::string_view mockarchive    = "mockarchive";
+        const std::string_view nonblocking    = "nonblocking";
+        const std::string_view passthrough    = "passthru";
+        const std::string_view random         = "random";
+        const std::string_view replication    = "replication";
+        const std::string_view struct_file    = "structfile";
+        const std::string_view universal_mss  = "univmss";
+        const std::string_view unixfilesystem = "unixfilesystem";
+        // clang-format on
+    } // namespace resource_type
+
+    enum class resource_status
+    {
+        up,
+        down,
+        auto_detect,
+        unknown
+    };
+
+    class resource_info
+    {
+    public:
+        // clang-format off
+        auto id() const noexcept -> const std::string&             { return id_; }
+        auto name() const noexcept -> const std::string&           { return name_; }
+        auto type() const noexcept -> const std::string&           { return type_; }
+        auto zone_name() const noexcept -> const std::string&      { return zone_name_; }
+        auto host_name() const noexcept -> const std::string&      { return host_name_; }
+        auto vault_path() const noexcept -> const std::string&     { return vault_path_; }
+        auto status() const noexcept -> resource_status            { return status_; }
+        auto context_string() const noexcept -> const std::string& { return context_string_; }
+        auto comments() const noexcept -> const std::string&       { return comments_; }
+        auto parent_id() const noexcept -> const std::string&      { return parent_id_; }
```

Comment:
please consider a handy lookup for parent_name() as well - could be best elsewhere.

Reply:
done.